native/src/components/Header.tsx

```diff
@@ -178,14 +194,18 @@ const Header = (props: PropsType): ReactElement => {
           )}
         </HorizontalLeft>
         <MaterialHeaderButtons cancelLabel={t('cancel')} theme={theme}>
-          {!peeking && categoriesAvailable && renderItem(t('search'), 'always', goToSearch, t('search'), 'search')}
-          {!peeking &&
-            goToLanguageChange &&
-            renderItem(t('changeLanguage'), 'always', goToLanguageChange, t('changeLanguage'), 'language')}
-          {showShare && renderItem(t('share'), 'never', onShare, t('share'), undefined)}
-          {showChangeLocation && renderItem(t('changeLocation'), 'never', goToLanding, t('changeLocation'), undefined)}
-          {renderItem(t('settings'), 'never', goToSettings, t('settings'), undefined)}
-          {routeCityModel && renderItem(t('disclaimer'), 'never', goToDisclaimer, t('disclaimer'), undefined)}
+          {renderItem(HeaderButtonTitle.Search, true, goToSearch, 'search', !peeking && categoriesAvailable)}
+          {renderItem(
+            HeaderButtonTitle.Language,
+            true,
+            goToLanguageChange,
+            'language',
+            !peeking && !!goToLanguageChange
+          )}
```

Comment:
Do these buttons appear at different times? Maybe we should just show/hide them together, i.e using the check `const visible = !peeking && categoriesAvailable && !!goToLanguageChange`. Not sure about that though, I have also never seen one button appear before another I think (at least not noticeable).

Reply:
as far is see it was not noticeable at all